### PR TITLE
refactor: clean up and align UI components related to CurrencyAmountPreview

### LINKED ISSUE #2429: Create a generic component TradeConfirmationModal (to replace LimitOrdersConfirmModal)
image

For Advanced orders, we will need a component for the confirmation modal.
Need to refactor `LimitOrdersConfirmModal` to create a new universal solution for confirming any trade.

### PR DESCRIPTION
**Pull request by shoom3301**
# Summary

Fixes #2429

This is mostly auto-refactoring, some components are moved to `common/pure` from `swap` and `limit-orders` modules.  
This PR is a preparation for refactoring of `LimitOrdersConfirmModal`.

## Changes
 - [refactor: move CurrencyLogo to common/pure](https://github.com/cowprotocol/cowswap/pull/2563/commits/5d113939fb0404a2ea1e030ce401ddce5a7d9014)
 - [refactor: move CurrencyLogo hooks](https://github.com/cowprotocol/cowswap/pull/2563/commits/24190cf6c7e7efba3e4434b99cd95fdd4a5f3b2a)
 - [refactor: move CurrencySelectButton](https://github.com/cowprotocol/cowswap/pull/2563/commits/d8307358a98263c70ce2cfd07494f5b9b261e30e)
 - [refactor: move CurrencyAmountPreview](https://github.com/cowprotocol/cowswap/pull/2563/commits/e49ba4699c5ae1f039a156f67ac4e330b78ea9e5)
 - [refactor: clean up FiatValue props](https://github.com/cowprotocol/cowswap/pull/2563/commits/9b2b38dfc4dc8714522d706a5e684115a81f4b38)
 - [refactor: extract PriceImpactIndicator component](https://github.com/cowprotocol/cowswap/pull/2563/commits/73ed8ad9b3f4fa54e71bcfffb664d1593e43f75b)
 - [refactor: move FiatValue and PriceImpactIndicator to common pure](https://github.com/cowprotocol/cowswap/pull/2563/commits/81da9233b3d4f209a38d6379045745b33d8acb2c)
 - [refactor: remove CurrencyInputPanelMod.tsx](https://github.com/cowprotocol/cowswap/pull/2563/commits/047b06eebaae241c34b8eb86b8f8a8a845625d84)
 - [refactor: align LimitOrdersConfirm imports](https://github.com/cowprotocol/cowswap/pull/2563/commits/65153b84ffcddb47369925eca3979db21bf3a858)

### CODE REVIEW

File: src/common/pure/CurrencyInputPanel/CurrencyInputPanel.tsx [hunk 1]
```diff
@@ -154,12 +153,7 @@ export function CurrencyInputPanel(props: CurrencyInputPanelProps) {
           </div>
           <div>
             <styledEl.FiatAmountText>
-              <FiatValue
-                isLoading={isRateLoading}
-                priceImpactLoading={priceImpactLoading}
-                fiatValue={fiatAmount}
-                priceImpact={priceImpact}
-              />
+              <FiatValue priceImpactParams={priceImpactParams} fiatValue={isRateLoading ? null : fiatAmount} />
```
Comment (shoom3301) [inline src/common/pure/CurrencyInputPanel/CurrencyInputPanel.tsx:156]: There was a single usage of `isLoading` prop. Essentially, `isLoading` is the same with `!!fiatAmount`

File: src/legacy/components/swap/styleds.tsx [hunk 2]
```diff
@@ -168,3 +169,91 @@ export const ResponsiveTooltipContainer = styled(TooltipContainer)<{ origin?: st
     transform-origin: ${origin ?? 'top left'};
   `}
 `
+// TODO: refactor these styles
+export const AuxInformationContainer = styled.div<{
```
Comment (shoom3301) [inline src/legacy/components/swap/styleds.tsx:173]: Was moved from `CurrencyInputPanelMod.tsx`

File: src/common/pure/CurrencyLogo/hooks/useCurrencyLogoURIs.ts [hunk 3]
```diff
@@ -3,10 +3,10 @@ import { SupportedChainId } from '@cowprotocol/cow-sdk'
 
 import EthereumLogo from 'legacy/assets/images/ethereum-logo.png'
 import XDaiLogo from 'legacy/assets/cow-swap/xdai.png'
-import { useExternalTokenLogo } from 'common/hooks/useExternalTokenLogo'
 import { NATIVE_CURRENCY_BUY_ADDRESS } from 'legacy/constants'
 import uriToHttp from 'lib/utils/uriToHttp'
 import { ADDRESS_IMAGE_OVERRIDE } from 'legacy/constants/tokens'
+import { useProxyTokenLogo } from '../../../../api/proxy'
```
Comment (nenadV91) [inline src/common/pure/CurrencyLogo/hooks/useCurrencyLogoURIs.ts:9]: Lets use absolute path here.

Comment (anxolin) [inline src/common/pure/CurrencyLogo/hooks/useCurrencyLogoURIs.ts:9]: why not absolute import?

File: src/legacy/components/CurrencyInputPanel/FiatValue/FiatValueMod.tsx [hunk 4]
```diff
@@ -1,67 +0,0 @@
...
-import { warningSeverity } from 'legacy/utils/prices'
-import { MouseoverTooltip } from 'legacy/components/Tooltip'
-
-// MOD imports
-// mod
-import Loader from 'legacy/components/Loader'
-import { formatPercent } from 'utils/amountFormat'
-import { FiatAmount } from 'common/pure/FiatAmount'
-
-export function FiatValue({
```
Comment (anxolin) [inline src/legacy/components/CurrencyInputPanel/FiatValue/FiatValueMod.tsx:17]: All good, not need to addres, but its a pitty git don't detected this as a move (shows as file added and removed)

not sure if `git mv` would have helped here. Or do one commit with the moving of the file, another with the changes

File: src/common/pure/CurrencyLogo/hooks/useCurrencyLogoURIs.ts [hunk 5]
```diff
@@ -42,9 +42,9 @@ function getTokenLogoURI(address: string, chainId: SupportedChainId = SupportedC
 const currencyLogoCache = new Map<string, Array<string>>()
 // TODO: must be refactored
 export default function useCurrencyLogoURIs(currency?: Currency | null): string[] {
-  const externalLogo = useExternalTokenLogo(currency)
-  // There is a modification of Token in useDetectNativeToken()
   const currencyAddress = currency ? (currency.isNative ? NATIVE_CURRENCY_BUY_ADDRESS : currency.address) : null
+  const externalLogo = useProxyTokenLogo(currency?.chainId, currencyAddress)
+  // There is a modification of Token in useDetectNativeToken()
```
Comment (nenadV91) [inline src/common/pure/CurrencyLogo/hooks/useCurrencyLogoURIs.ts:47]: Is this comment relevant to your line 46?

File: src/common/pure/CurrencyLogo/hooks/useCurrencyLogoURIs.ts (hunk 3 above)
Comment (shoom3301) [reply src/common/pure/CurrencyLogo/hooks/useCurrencyLogoURIs.ts:9]: My IDE generates it. It's a reason why I added the new rule to eslint :)

File: src/legacy/components/CurrencyInputPanel/FiatValue/FiatValueMod.tsx (hunk 4 above)
Comment (shoom3301) [reply src/legacy/components/CurrencyInputPanel/FiatValue/FiatValueMod.tsx:17]: Actually, I moved them using `git mv`. I guess it doesn't detect it properly because there are lots of changes

File: src/common/pure/CurrencyLogo/hooks/useCurrencyLogoURIs.ts (hunk 5 above)
Comment (shoom3301) [reply src/common/pure/CurrencyLogo/hooks/useCurrencyLogoURIs.ts:47]: Fixed

**Review by nenadV91 — APPROVED** [review]
